rudra/data_store/bigquery/base.py

```diff
@@ -15,27 +15,51 @@
 class BigqueryBuilder:
     """BigqueryBuilder class Implementation."""
 
-    def __init__(self, query_job_config=None, credential_path=None):
+    def __init__(self, query_job_config=None):
         """Initialize the BigqueryBuilder object."""
-        self.original_credential_path = os.getenv('GOOGLE_APPLICATION_CREDENTIALS') \
-                               or credential_path
-
-        try:
-            json.loads(self.original_credential_path)
-            json_credentials = True
-        except Exception as e:
-            logger.error("Not JSON credentials, reverting to local env JSON file: {}".format(e))
-            json_credentials = False
-
-        if json_credentials:
-            tfile = tempfile.NamedTemporaryFile(mode='w+', delete=False)
-            tfile.write(self.original_credential_path)
-            tfile.flush()
-            tfile.seek(0)
-            self.new_credential_path = tfile.name
-            os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = self.new_credential_path
-        else:
-            self.new_credential_path = self.original_credential_path
+        logger.info("Creating BigQuery Auth Credentials")
```

Comment:
Why are we creating the BQ auth credentials here? I think the previous approach of reading it from a JSON file was better. If you want to have a look, as to how to use GCP credentials as secrets in openshift, PTAL: https://cloud.google.com/blog/products/gcp/getting-started-with-red-hat-openshift-on-google-cloud-platform.

Reply:
Hey @abs51295 yeah I know that's how the previous version was when I implemented it, but based on the way secrets are stored in staging and prod, this is the best way to make use of them based on a review I did with @miteshvp yesterday.

Reply:
Well, we can store JSON file as a secret on staging and prod. I will suggest to go that way, since that's easier to read and maintain. But, if we have decided to go this way, then I will definitely like the credential generation part as a separate function, since the `__init__()` method currently looks very bloated.

Reply:
Created  a separate function now that is a good idea. We can do the JSON thing later maybe if needed since the credentials already exist on stage\prod.

Reply:
Yep this is done! The JSON thing maybe we can look at later on considering the keys right now already exist in staging\prod which is why we are changing the current code in the PR to use them when deployed on staging

Reply:
Will go with this approach since we have the secrets already in staging and prod.